api/api.go

```diff
@@ -23,13 +60,17 @@ type BlockChainAPI struct{}
 // wasn't synced up to a block where EIP-155 is enabled, but this behavior caused issues
 // in CL clients.
 func (api *BlockChainAPI) ChainId() *hexutil.Big {
-	return (*hexutil.Big)(big.NewInt(777))
+	return (*hexutil.Big)(api.config.ChainID)
 }
 
 // eth_blockNumber
 // BlockNumber returns the block number of the chain head.
 func (api *BlockChainAPI) BlockNumber() hexutil.Uint64 {
-	return hexutil.Uint64(65848272)
+	latestBlockHeight, err := api.Store.LatestBlockHeight(context.Background())
+	if err != nil {
+		return hexutil.Uint64(0)
```

Comment:
I wonder if it's worth panicking at this point or if nothing else logging an error, the way it's written now it might cause issues that will go unnoticed. If logging is not yet part of this PR maybe add TODO but also consider panic.

Reply:
Good idea :+1: 
I should definitely add a logger instance to `BlockChainAPI`.
For now, I have added a TODO and a panic in https://github.com/onflow/flow-evm-gateway/pull/11/commits/df9553c9bb149b730f438f84a92f24b5f121df1f